src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -181,7 +187,15 @@ def process(
         )
 
         collected_output = []
+        collected_sql_files: List[Path] = []
+        resolved_generated_root = resolve_without_follow(self.generated_root)
```

Comment:
nit: I still think we should make it an invariant that Paths passed into this class are resolved upfront, making this unnecessary. If you want, we can assert that all provided paths are absolute at least in the ctor.

Reply:
If you feel it's absolutely necessary, but really, relying on CWD is broken anyway, so passing the deploy / project roots around as resolved, absolute directories is really the only contract that makes sense to me. I don't like the idea that a class could silently just infer something wrong from the CWD if we ever mess up the logic. I'd much rather have the code fail during development, hence the suggestion to use assertions. Strong contracts make code easy to maintain.

Reply:
Ack, the changes are coming in the next commit. I have limited path resolution requirement to processors, not to the compiler. If we discover further use of resolved paths elsewhere in the compiler, we can then refactor the paths to the compiler itself be resolved.